presto-plugin-toolkit/src/test/java/io/prestosql/plugin/base/security/TestFileBasedSystemAccessControl.java

```diff
@@ -564,17 +565,66 @@ public void testSchemaOperations()
     }
 
     @Test
-    public void testCatalogOperations()
-    {
-        SystemAccessControl accessControl = newFileBasedSystemAccessControl("catalog.json");
-
-        assertEquals(accessControl.filterCatalogs(new SystemSecurityContext(admin, queryId), allCatalogs), allCatalogs);
-        Set<String> aliceCatalogs = ImmutableSet.of("open-to-all", "alice-catalog", "all-allowed");
-        assertEquals(accessControl.filterCatalogs(new SystemSecurityContext(alice, queryId), allCatalogs), aliceCatalogs);
-        Set<String> bobCatalogs = ImmutableSet.of("open-to-all", "all-allowed");
-        assertEquals(accessControl.filterCatalogs(new SystemSecurityContext(bob, queryId), allCatalogs), bobCatalogs);
-        Set<String> nonAsciiUserCatalogs = ImmutableSet.of("open-to-all", "all-allowed", "\u0200\u0200\u0200");
-        assertEquals(accessControl.filterCatalogs(new SystemSecurityContext(nonAsciiUser, queryId), allCatalogs), nonAsciiUserCatalogs);
+    public void testFilterCatalogs()
+    {
+        SystemAccessControl accessControl = newFileBasedSystemAccessControl("file-based-system-access-visibility.json");
+        Set<String> allCatalogs = ImmutableSet.of(
+                "alice-catalog",
+                "bob-catalog",
+                "specific-catalog",
+                "secret",
+                "hidden",
+                "open-to-all",
+                "blocked-catalog",
+                "unknown");
+
+        assertEquals(accessControl.filterCatalogs(ADMIN, allCatalogs), Sets.difference(allCatalogs, ImmutableSet.of("blocked-catalog")));
+        Set<String> aliceCatalogs = ImmutableSet.of("specific-catalog", "alice-catalog");
+        assertEquals(accessControl.filterCatalogs(ALICE, allCatalogs), aliceCatalogs);
+        Set<String> bobCatalogs = ImmutableSet.of("specific-catalog", "alice-catalog", "bob-catalog");
+        assertEquals(accessControl.filterCatalogs(BOB, allCatalogs), bobCatalogs);
+        Set<String> charlieCatalogs = ImmutableSet.of("specific-catalog");
+        assertEquals(accessControl.filterCatalogs(CHARLIE, allCatalogs), charlieCatalogs);
+    }
+
+    @Test
+    public void testSchemaRulesForCheckCanShowSchemas()
+    {
+        SystemAccessControl accessControl = newFileBasedSystemAccessControl("file-based-system-access-visibility.json");
```

Comment:
The thing that we load external file and run tests on it make tests very difficult to follow. Especially when file is reused between tests.

Best would be to have some kind of ddl to create file based system access control with few rules and then test them.